File: src/extension/compass/__tests__/fixtures/type_hints.cpp
class OrganizationService {
};

class Logger {
};

class Repository {
};

class TenantController {
    Logger logger;

public:
    TenantController(OrganizationService service) {
    }

    OrganizationService resolve(Repository repo) {
        return OrganizationService();
    }

    int count(int value) {
        return value;
    }
};

File: src/extension/compass/__tests__/fixtures/type_hints.rs
struct OrganizationService {}

struct Logger {}

struct Repository {}

struct TenantController {
    logger: Logger,
}

impl TenantController {
    fn new(service: OrganizationService) -> TenantController {
        TenantController { logger: Logger {} }
    }

    fn resolve(&self, repo: &Repository) -> OrganizationService {
        OrganizationService {}
    }

    fn count(&self, value: i32) -> String {
        String::new()
    }
}
